exporter/newrelicexporter/newrelic.go

```diff
@@ -78,79 +65,25 @@ func clientOptions(apiKey string, apiKeyHeader string, hostOverride string, inse
 	return options
 }
 
-func newTraceExporter(l *zap.Logger, c configmodels.Exporter) (*exporter, error) {
-	nrConfig, ok := c.(*Config)
-	if !ok {
-		return nil, fmt.Errorf("invalid config: %#v", c)
-	}
-
+func newExporter(l *zap.Logger, nrConfig endpointConfig, createFactory factoryBuilder) (*exporter, error) {
```

Comment:
Do you want to return a pointer or value here?

Reply:
If we change to a value type, that means we can no longer return `nil`. Is that preferable? 🤔